qemu/tests/rebase_negative_test_on_iscsi.py

```diff
@@ -0,0 +1,120 @@
+                                  self.params["image_name_%s" % image_name])
+            self.params.update(
+                {"image_name_%s" % image_name: filename,
+                 "image_size_%s" % image_name: self.params["lv_size"]})
+            t_params = self.params.object_params(image_name)
+            qemu_image = qemu_storage.QemuImg(t_params,
+                                              self.data_dir, image_name)
+            logging.info("Create image('%s') on %s." %
+                         (image_name, qemu_image.storage_type))
+            qemu_image.create(self.params)
+        error_context.context("Rebase snapshot to backingfile", logging.info)
+        for images in rebase_chain:
+            images = map(lambda x: x.strip(), images.split(">"))
+            try:
+                image = images[0]
+                base = images[1]
+            except IndexError:
+                msg = "Invalid format of'rebase_chain' params \n"
+                msg += "format like: 'image > base;image> base2;image2 > base2'"
+                break
+            self.params["image_chain"] = " ".join([base, image])
+            self.params["base_image_filename"] = image
+            try:
+                output = ""
+                t_params = self.params.object_params(image)
+                rebase_test = qemu_storage.QemuImg(t_params,
+                                                   self.data_dir, image)
+                cache_mode = self.params.get("cache_mode")
+                rebase_test.rebase(self.params, cache_mode)
+            except process.CmdError, err:
+                output = err.result.stderr
+                logging.info("Rebase image('%s') failed: %s." %
+                             (image, output))
+                if "(core dumped)" in output:
+                    msg = "qemu-img core dumped when change"
+                    msg += " image('%s') backing file to %s" % (image, base)
+                    break
+                trigger_negative = True
+            image_info = rebase_test.info()
+            if not image_info:
+                msg = "Fail to get image('%s') info" % image
+                break
+            backingfile = re.search(r'backing file: +(.*)', image_info, re.M)
+            base_image_name = rebase_test.base_image_filename
+            if not output:
+                if backingfile and base_image_name not in backingfile.group(0):
+                    msg = "Expected backing file: %s" % base_image_name
+                    msg += " Actual backing file: %s" % backingfile
+                    break
+                if not backingfile:
+                    msg = "Expected backing file: %s" % base_image_name
+                    msg += " Actual backing file is null!"
+                    break
+        self.lvmdevice.cleanup()
+        if msg:
+            raise error.TestFail(msg)
+        if not trigger_negative:
+            msg += "\nAll rebase operation('%s') passed" % rebase_chain
```

Comment:
rebase_chain contains all the rebase opeations, one of them should be failed as this script is used for negative test. For this case, rebase_chain only contains the failed rebase opeation.